test/cpp/qps/server_async.cc

```diff
@@ -166,8 +100,16 @@ class AsyncQpsServerTest {
         return;
       }));
     }
-    while (!got_sigint) {
-      std::this_thread::sleep_for(std::chrono::seconds(5));
+  }
+  ~AsyncQpsServerTest() {
+    server_->Shutdown();
+    srv_cq_.Shutdown();
```

Comment:
Ok, thanks for the clarification. This implies that Next will only return false after there are indeed no more events. So I'm on board. Merging shortly.